File: go/callee-service-gofr/callee/calleecontroller.go
package callee

import (
	"gofr.dev/pkg/gofr"
)

func RouteCallee(app *gofr.App) {
	app.GET("/callees/sayMyName", func(ctx *gofr.Context) (interface{}, error) {
		name := ctx.Param("name")
		return SayMyName(name), nil
	})

	app.GET("/callees/sayMyOtherName/{name}", func(ctx *gofr.Context) (interface{}, error) {
		name := ctx.PathParam("name")
		return SayMyOtherName(name), nil
	})

	/*
			app.POST("/callees/save", func(ctx *gofr.Context) (interface{}, error) {
				var callee Callee
		        if err := ctx.Bind(&callee); err != nil {
		            return nil, gofr.NewError(http.StatusBadRequest, "invalid request body")
		        }
				return Callee{ID: "0", Message: "Storing your message: " + callee.Message}
			})
	*/

}

File: python/callee-service-python/src/callee_service/controller/callee_controller.py
"""
Equivalent of CalleeController.kt

FastAPI router for /callees endpoints.
- GET  /callees/sayMyName?name=...      (query param)
- GET  /callees/sayMyOtherName/{name}   (path param)
- POST /callees/save                    (JSON body)

Security note: @RolesAllowed("standard_role") is reproduced via a
Depends() dependency that checks the Authorization header. Swap in
your real OIDC/JWT validation as needed.
"""

import logging

from fastapi import APIRouter, Depends
from fastapi.security import HTTPBearer

from callee_service.controller.dto.callee import Callee
from callee_service.logic.callee_logic import CalleeLogic

logger = logging.getLogger(__name__)

router = APIRouter(prefix="/callees", tags=["callees"])

_bearer_scheme = HTTPBearer(auto_error=False)


@router.get("/sayMyName", summary="Say my name")
async def say_my_name(
    name: str,
    logic: CalleeLogic = Depends(CalleeLogic),
) -> Callee:
    return await logic.say_my_name(name)


@router.get("/sayMyOtherName/{name}", summary="Say my other name")
async def say_my_other_name(
    name: str,
    logic: CalleeLogic = Depends(CalleeLogic),
) -> Callee:
    return await logic.say_my_other_name(name)


@router.post("/save", summary="Save a callee message")
async def save(
    callee: Callee,
    logic: CalleeLogic = Depends(CalleeLogic),
) -> Callee:
    return await logic.save(callee)
